packages/dds/tree/src/test/feature-libraries/default-schema/schemaChecker.spec.ts

```diff
@@ -0,0 +1,722 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { Multiplicity } from "../../../../dist/index.js";
+// Reaching into internal module just to test it
+import {
+	SchemaValidationErrors,
+	compliesWithMultiplicity,
+	isFieldInSchema,
+	isNodeInSchema,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../../feature-libraries/default-schema/schemaChecker.js";
+import {
+	FieldKinds,
+	type FlexFieldKind,
+	type FullSchemaPolicy,
+} from "../../../feature-libraries/index.js";
+import {
+	LeafNodeStoredSchema,
+	MapNodeStoredSchema,
+	ObjectNodeStoredSchema,
+	ValueSchema,
+	type FieldKey,
+	type FieldKindData,
+	type FieldKindIdentifier,
+	type MapTree,
+	type StoredSchemaCollection,
+	type TreeFieldStoredSchema,
+	type TreeNodeSchemaIdentifier,
+	type TreeNodeStoredSchema,
+	type Value,
+} from "../../../core/index.js";
+import { brand } from "../../../util/index.js";
+import type { IFluidHandle, IRequest, IResponse } from "@fluidframework/core-interfaces";
+
+/**
+ * A fake FluidHandle to be used in tests.
+ * It's completely non-functional other than the `IFluidHandle` property, which is required
+ * for the validations that SharedTree does on a FluidHandle value in a node.
+ */
+class TestFluidHandle implements IFluidHandle {
```

Comment:
Can you use MockHandle from test-runtime-utils instead?

Reply:
Yep, updated